lib/std/os/bits/linux.zig

```diff
@@ -88,11 +88,29 @@ pub const FUTEX_PRIVATE_FLAG = 128;
 
 pub const FUTEX_CLOCK_REALTIME = 256;
 
-pub const PROT_NONE = 0;
-pub const PROT_READ = 1;
-pub const PROT_WRITE = 2;
-pub const PROT_EXEC = 4;
+/// page can not be accessed
+pub const PROT_NONE = 0x0;
+
+/// page can be read
+pub const PROT_READ = 0x1;
+
+/// page can be written
+pub const PROT_WRITE = 0x2;
+
+/// page can be executed
+pub const PROT_EXEC = 0x4;
+
+/// page may be used for atomic ops
+pub const PROT_SEM = switch (builtin.arch) {
+    // TODO: also xtensa
+    .mips, .mipsel, .mips64, .mips64el => 0x10,
+    else => 0x8,
+}
+
+/// mprotect flag: extend change to start of growsdown vma
```

Comment:
- What's VMA?
- What change? You mean the block being modified in the call to `protect`?

Reply:
This is the comment directly out of the kernel headers.

See the man page for more info, `mprotect(2)`:
>  PROT_GROWSUP
>                  Apply the protection mode up to the end of a mapping that grows upwards.  (Such mappings are created for the stack area on  architectures for example, HP-PARISC—that have an upwardly growing stack.)
> 
>  PROT_GROWSDOWN
>                  Apply the protection mode down to the beginning of a mapping that grows downward (which should be a stack segment or a segment mapped with the MAP_GROWSDOWN flag set).

Reply:
I'm not sure those are better unless there's some sort of point to the direction that the protection is set in - whatever that means.
Like, we're just dealing with regions of memory here - sure you might use them by going up or down - but if that's what it's about, shouldn't it just say that? 😁